homeassistant/components/media_player/yamaha.py

```diff
@@ -50,7 +50,10 @@ def update(self):
         self._muted = self._receiver.mute
         self._volume = (self._receiver.volume/100) + 1
         self._current_source = self._receiver.input
-        self._source_list = list(self._receiver.inputs().keys())
+
+        if self.source_list is None:
+            self._source_list = list(self._receiver.inputs().keys())
```

Comment:
Nope, since additional mapping might happen i can not sort the return value from the device itself